### ISSUE TITLE
BindableNumber doesn't respect min/max range when rounding value by precision

### ISSUE THREAD

**Issue by frenzibyte** [ppy/osu-framework#4129]
Noticed while I was doing something else, thought reporting it would be a better choice rather than leaving it and potentially forgetting about it.

Setting a certain precision and a minimum/maximum range beyond that precision causes `Value` to potentially become out of range, due to the final value computation logic not respecting ranges when rounding.

For example, a bindable with `0.5` precision and minimum range and `Value` set to `10.25`, results in the bindable having `10.5` instead of `10.0`:
```csharp
var number = new BindableDouble();
number.Precision = 0.5;
number.MinValue = 10.25;
number.Value = 10.25;
// number.Value is 10.0, instead of being 10.5
```

**Comment by bdach** [comment]
Is there an actual reason why anybody would do this? Maybe values of `{Min,Max}Value` that aren't multiplies of `Precision` should be straight banned?

**Comment by frenzibyte** [comment]
Yeah was gonna suggest that, both should just respect precision as well
